web/src/App.tsx

```diff
@@ -34,15 +33,17 @@ export default class App extends React.Component<Props, State> {
         this.componentDidMount = this.componentDidMount.bind(this);
     }
 
-    componentDidMount() {
+    //add event listener for popstate so that when back button or forward button 
+    //is clicked, the state changes accordingly (basically to track button clicks)
```

Comment:
```suggestion
    // add event listener for popstate so that when back button or forward button 
    // is clicked update react state to show results / upload file button
```